language-plutus-core/src/Language/PlutusCore/Renamer.hs

```diff
@@ -83,39 +81,39 @@ runScopedRenameM = runRenameM emptyScopedUniquesRenaming
 
 -- | The empty 'UniquesRenaming'.
 emptyUniquesRenaming :: UniquesRenaming unique
-emptyUniquesRenaming = UniquesRenaming mempty
+emptyUniquesRenaming = mempty
 
 -- | The empty 'ScopedUniquesRenaming'.
 emptyScopedUniquesRenaming :: ScopedUniquesRenaming
```

Comment:
derive monoid for ScopedUniquesRenaming and just use `mempty`?

Reply:
Every derived instance results in a certain slowdown (though, `Semigroup` and `Monoid` are probably more or less benign compared to `Show` and `Ord`, there was a blog post about this showing some not too pleasant numbers, but I can't find it) and also has a certain semantic overhead ("so why do you need `Semigroup` and `Monoid` here?"). Since `ScopedUniquesRenaming` is local and not exported, I'd prefer to avoid defining instances for it when it's that easy.

But that's not too important, I can derive `Monoid` if that's what you prefer.

Reply:
As a reader, I find extra top-level definitions to be more annoying. And at the use site `mempty` very quickly tells me its just the same empty value as usual.

I would hope newtype derivings are quick!

In addition, using monoid for things like this is so common that my immediate reaction is "hm, I wonder if there's a reason this isn't a monoid" ;) I think we could really do with `deriving all` for newtypes!

Reply:
Ok, this is convincing, but how do I derive the monoid instance of `ScopedUniquesRenaming`? It's a `data`, not a `newtype`.

Reply:
This is not a big deal anyhow.

Reply:
> Horrible suggestion: `newtype ScopedUniquesRenaming = ScopedUniquesRenaming (UniquesRenaming TypeUnique, UniquesRenaming TermUnique)`

Thanks, no :)